### ISSUE TITLE
Reverse Proxy / Load Balancer in Front of docker-letsencrypt-nginx-proxy-companion

### ISSUE THREAD

**Issue by netlander** [nginx-proxy/nginx-proxy#1451]
Is it possible to deploy an architecture that spans multiple physical servers as depicted in the diagram below?

I should imagine that the public facing nginx Docker container would be a plain nginx proxy that just relays traffic to backend containers.

Any pointers in the right direction highly appreciated.

![image](https://user-images.githubusercontent.com/4721044/84575032-490b3580-ada2-11ea-885d-024d0ed62e89.png)

**Comment by azlux** [comment]
You want a TCP load-balancer, not a reverse proxy.
You cannot use a nginx in from of all other because of the HTTPS on each server-0X.

**Comment by netlander** [comment]
In my use case there's a need for the load balancer to proxy requests to various domains to particular physical servers, for ex. www.abcd.com to server-01, www.efgh.com to server-02, etc...

Not sure a TCP load balancer is the solution but will take a look.

Do you have any pointers to documentation or specific TCP load balancer worth mentioning?

**Comment by dromer** [comment]
Have a look at haproxy for doing this.

**Comment by brezhitskiy42** [comment]
@netlander Hi. Did you found the solution?

**Comment by netlander** [comment]
@brezhitskiy42 Not yet, but still looking for something that works. 
@dromer has suggested haproxy which looks promising but haven looked into it in any depth.
Will update this thread when I get to a working solution.

**Comment by netlander** [comment]
@brezhitskiy42 I have abandoned the pursuit of Haproxy as I fond that it's not suitable for an opinionated setup like this one here.

Instead I went for a [Three Step Proxied Container](https://github.com/nginx-proxy/docker-letsencrypt-nginx-proxy-companion#step-3---proxied-containers) and manually edited `nginx.conf` to do the forwarding.

This solution is something of a hack but for my use-case this approach worked well.
